mix.exs

```diff
@@ -339,96 +339,125 @@ defmodule EMQXUmbrella.MixProject do
-          emqx_enterprise: :load,
-          emqx_bridge_kafka: :permanent,
-          emqx_bridge_pulsar: :permanent,
-          emqx_bridge_gcp_pubsub: :permanent,
-          emqx_bridge_cassandra: :permanent,
-          emqx_bridge_opents: :permanent,
-          emqx_bridge_clickhouse: :permanent,
-          emqx_bridge_dynamo: :permanent,
-          emqx_bridge_hstreamdb: :permanent,
-          emqx_bridge_influxdb: :permanent,
-          emqx_bridge_iotdb: :permanent,
-          emqx_bridge_matrix: :permanent,
-          emqx_bridge_mongodb: :permanent,
-          emqx_bridge_mysql: :permanent,
-          emqx_bridge_pgsql: :permanent,
-          emqx_bridge_redis: :permanent,
-          emqx_bridge_rocketmq: :permanent,
-          emqx_bridge_tdengine: :permanent,
-          emqx_bridge_timescale: :permanent,
-          emqx_bridge_sqlserver: :permanent,
-          emqx_oracle: :permanent,
-          emqx_bridge_oracle: :permanent,
-          emqx_bridge_rabbitmq: :permanent,
-          emqx_schema_registry: :permanent,
-          emqx_eviction_agent: :permanent,
-          emqx_node_rebalance: :permanent,
-          emqx_ft: :permanent
-        ],
-        else: [
-          emqx_telemetry: :permanent
-        ]
-      )
+        if enable_quicer?() do
+          [:quicer]
+        else
+          []
+        end ++
+        if enable_bcrypt?() do
+          [:bcrypt]
+        else
+          []
+        end ++
+        if enable_jq?() do
+          [:jq]
+        else
+          []
+        end ++
+        if(is_app(:observer),
+          do: [:observer],
+          else: []
+        ) ++
+        case edition_type do
+          :enterprise ->
+            [
+              :emqx_license,
+              :emqx_enterprise,
+              :emqx_bridge_kafka,
+              :emqx_bridge_pulsar,
+              :emqx_bridge_gcp_pubsub,
+              :emqx_bridge_cassandra,
+              :emqx_bridge_opents,
+              :emqx_bridge_clickhouse,
+              :emqx_bridge_dynamo,
+              :emqx_bridge_hstreamdb,
+              :emqx_bridge_influxdb,
+              :emqx_bridge_iotdb,
+              :emqx_bridge_matrix,
+              :emqx_bridge_mongodb,
+              :emqx_bridge_mysql,
+              :emqx_bridge_pgsql,
+              :emqx_bridge_redis,
+              :emqx_bridge_rocketmq,
+              :emqx_bridge_tdengine,
+              :emqx_bridge_timescale,
+              :emqx_bridge_sqlserver,
+              :emqx_oracle,
+              :emqx_bridge_oracle,
+              :emqx_bridge_rabbitmq,
+              :emqx_schema_registry,
+              :emqx_eviction_agent,
+              :emqx_node_rebalance,
+              :emqx_ft
+            ]
+
+          _ ->
+            [:emqx_telemetry]
+        end
+
+    system_apps ++
+      for app <- db_apps do
+        {app, :load}
+      end ++
+      [emqx_machine: :permanent] ++
+      for app <- business_apps do
+        {app, :load}
+      end
```

Comment:
```suggestion
      Enum.map(business_apps, & {&1, :load})
```

Although, this seems to change most apps from `:permanent` to `load` only. :thinking: 
Shouldn't those be permanent?

Reply:
Yes, this is the intention. Init script should not start any apps that depend on mria, because it causes mria to start before ekka initializes the callbacks and before `mgmt_cli` registers the commands. 
The apps will be started by `emqx_machine` (so the sequence is the same as during join/leave)

Reply:
Yes, you are right.